File: GridWallManagement.App.Database/Common/BaseModel.cs
namespace GridWallManagement.App.Database.Common
{
    public class BaseModel : GuidBase
    {
        public bool IsActive { get; set; } = false;

        public string CreatedBy { get; set; }

        public DateTime CreatedDate { get; set; } = DateTime.UtcNow;

        public string? ModifiedBy { get; set; }

        public DateTime? ModifiedDate { get; set; }

        public void Update(string modifiedBy)
        {
            this.ModifiedBy = modifiedBy;
            this.ModifiedDate = DateTime.UtcNow;
        }

        public void Add(string createdBy)
        {
            this.CreatedBy = createdBy;
            this.CreatedDate = DateTime.UtcNow;
            IsActive = true;
        }

        public void Delete(string modifiedBy)
        {
            Update(modifiedBy);
            IsActive = false;
        }
    }
}

File: GridWallManagement.Web/src/app/shared/models/common/BaseModel.ts
export class BaseModel {
  id: string;
  createdBy: string;
  createdDate: Date;
  modifiedBy?: string;
  modifiedDate?: Date;

  constructor(
    id: string = '',
    createdBy: string = '',
    createdDate: Date = new Date(),
    modifiedBy?: string,
    modifiedDate?: Date
  ) {
    this.id = id;
    this.createdBy = createdBy;
    this.createdDate = createdDate;
    this.modifiedBy = modifiedBy;
    this.modifiedDate = modifiedDate;
  }
}
